Tor failing to startup

```
2022-10-13 22:04:56,288UTC INFO [VortexClient] Connecting to coordinator Taproot Testnet: y74gsfy6u7s73jl53x52gcaqv3b7f76rntevwymk2ovziytz5p35tiad.onion:12523
2022-10-13 22:04:56,356UTC INFO [VortexClient] Connecting to coordinator Lightning Testnet: tmojtwaiyowq6m5npl63v7cvb2asmodiindwztf7qramfdvre6qsfvid.onion:12523
2022-10-13 22:04:56,363UTC INFO [VortexClient] Connecting to coordinator Taproot Converter Testnet: sqwdsbcu3cd5i2gyjwgqgmf75yq6q7pwtfyiplrouvwrxlgfjqbhz3id.onion:12523
2022-10-13 22:04:56,389UTC ERROR [Daemon$] Error
java.lang.RuntimeException: Connection failed.
	at akka.http.impl.engine.ws.WebSocketClientBlueprint$UpgradeStage$1$$anon$1.onUpstreamFailure(WebSocketClientBlueprint.scala:138)
	at akka.stream.impl.fusing.GraphInterpreter.processEvent(GraphInterpreter.scala:525)
	at akka.stream.impl.fusing.GraphInterpreter.execute(GraphInterpreter.scala:390)
	at akka.stream.impl.fusing.GraphInterpreterShell.runBatch(ActorGraphInterpreter.scala:650)
	at akka.stream.impl.fusing.GraphInterpreterShell$AsyncInput.execute(ActorGraphInterpreter.scala:521)
	at akka.stream.impl.fusing.GraphInterpreterShell.processEvent(ActorGraphInterpreter.scala:625)
	at akka.stream.impl.fusing.ActorGraphInterpreter.akka$stream$impl$fusing$ActorGraphInterpreter$$processEvent(ActorGraphInterpreter.scala:800)
	at akka.stream.impl.fusing.ActorGraphInterpreter$$anonfun$receive$1.applyOrElse(ActorGraphInterpreter.scala:818)
	at akka.actor.Actor.aroundReceive(Actor.scala:537)
	at akka.actor.Actor.aroundReceive$(Actor.scala:535)
	at akka.stream.impl.fusing.ActorGraphInterpreter.aroundReceive(ActorGraphInterpreter.scala:716)
	at akka.actor.ActorCell.receiveMessage(ActorCell.scala:579)
	at akka.actor.ActorCell.invoke(ActorCell.scala:547)
	at akka.dispatch.Mailbox.processMailbox(Mailbox.scala:270)
	at akka.dispatch.Mailbox.run(Mailbox.scala:231)
	at akka.dispatch.Mailbox.exec(Mailbox.scala:243)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:387)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1311)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1840)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1806)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:177)
Caused by: akka.stream.StreamTcpException: Tcp command [Connect(127.0.0.1/<unresolved>:9050,None,List(),Some(10 seconds),true)] failed because of java.net.ConnectException: Connection refused
Caused by: java.net.ConnectException: Connection refused
	at java.base/sun.nio.ch.Net.pollConnect(Native Method)
	at java.base/sun.nio.ch.Net.pollConnectNow(Net.java:672)
	at java.base/sun.nio.ch.SocketChannelImpl.finishConnect(SocketChannelImpl.java:973)
	at akka.io.TcpOutgoingConnection$$anonfun$connecting$1.$anonfun$applyOrElse$4(TcpOutgoingConnection.scala:111)
	at akka.io.TcpOutgoingConnection.akka$io$TcpOutgoingConnection$$reportConnectFailure(TcpOutgoingConnection.scala:53)
	at akka.io.TcpOutgoingConnection$$anonfun$connecting$1.applyOrElse(TcpOutgoingConnection.scala:111)
	at akka.actor.Actor.aroundReceive(Actor.scala:537)
	at akka.actor.Actor.aroundReceive$(Actor.scala:535)
	at akka.io.TcpConnection.aroundReceive(TcpConnection.scala:33)
	at akka.actor.ActorCell.receiveMessage(ActorCell.scala:579)
	at akka.actor.ActorCell.invoke(ActorCell.scala:547)
	at akka.dispatch.Mailbox.processMailbox(Mailbox.scala:270)
	at akka.dispatch.Mailbox.run(Mailbox.scala:231)
	at akka.dispatch.Mailbox.exec(Mailbox.scala:243)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:387)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1311)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1840)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1806)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:177)
```

--- a/app/rpc-server/src/main/scala/com/lnvortex/rpc/Daemon.scala
+++ b/app/rpc-server/src/main/scala/com/lnvortex/rpc/Daemon.scala
@@ -39,6 +39,8 @@ object Daemon extends App with Logging {
 
   logger.info("Starting...")
 
+  val torF = config.clientConfig.torConf.start()
+
   val configStartF = serverConfig.start()
 
   val server = RpcServer(
@@ -59,6 +61,7 @@ object Daemon extends App with Logging {
   val clientF = configStartF.flatMap { _ =>
     for {
       _ <- clientManager.config.start()
+      _ <- torF
       _ <- clientManager.start()
     } yield ()
   }

--- a/core/src/main/resources/reference.conf
+++ b/core/src/main/resources/reference.conf
@@ -38,7 +38,8 @@ bitcoin-s {
         socks5 = "127.0.0.1:9050"
     }
     tor {
-        provided = true
+        use-random-ports = true
+        provided = false
         enabled = true
     }
     bitcoind-rpc = ${vortex.bitcoind}

